src/hotspot/share/prims/jvm.cpp

```diff
@@ -627,7 +627,6 @@ JVM_END
 
 JVM_ENTRY(void, JVM_MonitorWait(JNIEnv* env, jobject handle, jlong ms))
   Handle obj(THREAD, JNIHandles::resolve_non_null(handle));
-  JavaThreadInObjectWaitState jtiows(thread, ms != 0);
```

Comment:
This is still changing the behaviour for non-virtual threads.

Reply:
Thanks. Yes, that is right. Will make one more update.